vhdbuilder/packer/init-variables.sh

```diff
@@ -264,7 +289,8 @@ cat <<EOF > vhdbuilder/packer/settings.json
   "windows_image_version": "${WINDOWS_IMAGE_VERSION}",
   "imported_image_name": "${IMPORTED_IMAGE_NAME}",
   "sig_image_name":  "${SIG_IMAGE_NAME}",
-  "arm64_os_disk_snapshot_name": "${ARM64_OS_DISK_SNAPSHOT_NAME}"
+  "arm64_os_disk_snapshot_name": "${ARM64_OS_DISK_SNAPSHOT_NAME}",
+  "os_disk_size_gb": "${OS_DISK_SIZE_GB}"
```

Comment:
Will `os_disk_size_gb` as user variable or `OS_DISK_SIZE_GB` as environment variable be used?

Reply:
The latter will be used. I think in any way, OS_DISK_SIZE_GB is already an environment variable in this file and will be passed to windows-vhd-builder.json. "os_disk_size_gb" is just an optional variable used in Azure Arm Builder [here](https://www.packer.io/plugins/builders/azure/arm) In that json, the default user name for "{{user `os_disk_size_gb`}}" in "builders" is packer, right? 

I think if user specifies OS_DISK_SIZE_GB, it will be set as an environment variable and passed to run-packer-windows in "Building windows VHD". (.builder...) If init-variables uses the values in windows-image.env, ${OS_DISK_SIZE_GB} will then be overridden.

Reply:
If OS_DISK_SIZE_GB is used, you do not need to add this line

Reply:
In [this previous commit](https://github.com/Azure/AgentBaker/commit/5827cf46b5497ee52ef9766b458c8064b033ddec), this line was not added. If we see the pipeline [here](https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=55605850&view=logs&j=26d3b11b-de37-5bdb-7f8b-686d226a8fb0&t=73630cad-3f30-5fb3-16e6-4f2994f8a555), the log outputs "No need to expand the OS partition size" because line 81 of configure-windows-vhd finds $customizedDiskSize is empty. This was set in windows-vhd-builder.json as user `os_disk_size_gb`.

I guess the way it works is if there is a field "os_disk_size_gb" in settings.json, windows-vhd-builder.json will use it? 

There are some common names in settings.json and "variables" in windows-vhd-builder.json?

Reply:
```There are some common names in settings.json and "variables" in windows-vhd-builder.json?``` I think that there was some misunderstanding in adding these variables before. I think that environment variables cannot be set when the variable is added into settings.json.

Reply:
Since the variable "os_disk_size_gb" is added into settings.json, [this](https://github.com/Azure/AgentBaker/pull/1827/commits/3c8689234125da8ce56421362f02e241269dd6c9) removes the env variable in windows-vhd-builder. There is no need to set environment variable in .builder-release-template-windows, so  "-e OS_DISK_SIZE_GB" is removed [here](https://github.com/Azure/AgentBaker/pull/1827/commits/126ea556374e8ac466a90793252586553d4874da).
To see if it works, size for 2019, 2019 containerd, 2022 containerd were set to 32GB, not set, 40GB in windows-image.env.
The log is [here](https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=55732428&view=results).

For 2019, building windows vhd outputs "Customized OS disk size is 32 GB".
For 2019 containerd, nothing was shown since the variable was not set.
For 2022 containerd, building outputs "Customized OS disk size is 40 GB".